docs/CodingGuidelines.md

```diff
@@ -0,0 +1,174 @@
+
```

Comment:
General comment on whole doc:

I think the separation of topics could be better. If I want to review the guidelines for testing, I can't go to a "testing" section. I have to kinda skim through the whole doc for it. Same with pull requests.

Could we group the items by topic? "Testing", "Pull Requests", "Syntax/Linting", "Coding Best Practices", etc.

Reply:
Yes i think that is a good idea, i'd like to have that done in a future revision